openpaas-james/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/openpaas/james/common/LinagoraKeystoreSetMethodContract.scala

```diff
@@ -0,0 +1,489 @@
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |  "sessionState": "${SESSION_STATE.value}",
+         |  "methodResponses": [[
+         |    "Keystore/set",
+         |    {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "notCreated": {
+         |        "K87": {
+         |          "type": "invalidArguments",
+         |          "description": "Some server-set properties were specified",
+         |          "properties": ["id"]
+         |        }
+         |      }
+         |    },
+         |    "c1"]]
+         |}""".stripMargin)
+  }
+
+  @Test
+  def keystoreSetCreateShouldBeIdempotent(): Unit = {
+    val request = s"""{
+                     |	"using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:pgp"],
+                     |	"methodCalls": [
+                     |		["Keystore/set", {
+                     |			"accountId": "$ACCOUNT_ID",
+                     |			"create": {
+                     |        "K87": {
+                     |          "key": "$PGP_KEY_ARMORED"
+                     |        },
+                     |        "K88": {
+                     |          "key": "$PGP_KEY_ARMORED"
+                     |        }
+                     |      }
+                     |		}, "c1"]
+                     |	]
+                     |}""".stripMargin
+
+    val response = `given`()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .log().ifValidationFails()
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |	"sessionState": "${SESSION_STATE.value}",
+         |	"methodResponses": [
+         |		[
+         |			"Keystore/set",
+         |			{
+         |				"accountId": "$ACCOUNT_ID",
+         |				"created": {
+         |          "K87": {
+         |            "id": "$PGP_KEY_ID"
+         |          },
+         |          "K88": {
+         |            "id": "$PGP_KEY_ID"
+         |          }
+         |				}
+         |			}, "c1"]
+         |	]
+         |}""".stripMargin)
+  }
+
+  @Test
+  def sessionShouldAdvertiseCapability(): Unit = {
+    val sessionJson: String = `given`()
+    .when()
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .headers(getHeadersWith(BOB_BASIC_AUTH_HEADER))
+      .get("/session")
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(sessionJson)
+      .inPath("capabilities.com:linagora:params:jmap:pgp")
+      .isEqualTo("{}")
+  }
+}
```

Comment:
Shouldn't we assert the content of the saved key(s) (by using probe, repository...) and replace the assertion with Keystore/get when it is available later?